flexget/plugins/input/kitsu.py

```diff
@@ -0,0 +1,114 @@
+            'limit_to': {'type': 'string', 'enum': ['airing', 'finished']}
+        },
+        'required': ['username'],
+        'additionalProperties': False,
+    }
+
+    @cached('kitsu', persist='2 hours')
+    def on_task_input(self, task, config):
+        entries = []
+        user_payload = {'filter[name]': config['username']}
+        try:
+            user_response = task.requests.get('https://kitsu.io/api/edge/users', params=user_payload)
+        except RequestException as e:
+            error_message = 'Error finding User url: {url}'.format(url=e.request.url)
+            if hasattr(e, 'response'):
+                error_message += ' status: {status}'.format(status=e.response.status_code)
+            log.debug(error_message, exc_info=True)
+            raise plugin.PluginError(error_message)
+        user = user_response.json()
+        if len(user['data']) < 1:
+            raise plugin.PluginError('no such username found "{name}"'.format(name=config['username']))
+        next_url = 'https://kitsu.io/api/edge/users/{id}/library-entries'.format(id=user['data'][0]['id'])
+        payload = {'filter[status]': ','.join(config['lists']), 'filter[media_type]': 'Anime', 'include': 'media',
+                   'page[limit]': 20}
+        try:
+            response = task.requests.get(next_url, params=payload)
+        except RequestException as e:
+            error_message = 'Error getting list from {url}'.format(url=e.request.url)
+            if hasattr(e, 'response'):
+                error_message += ' status: {status}'.format(status=e.response.status_code)
+            log.debug(error_message, exc_info=True)
+            raise plugin.PluginError(error_message)
+
+        while response:
+            json_data = response.json()
+
+            for item, anime in zip(json_data['data'], json_data['included']):
+                if item['relationships']['media']['data']['id'] != anime['id']:
+                    raise ValueError(
+                        'Anime IDs {id1} and {id2} do not match'.format(
+                            id1=item['relationships']['media']['data']['id'], id2=anime['id']))
+                limit_to = config.get('limit_to')
+                if limit_to is not None:
+                    if limit_to == 'airing' and anime['attributes']['endDate'] is not None:
+                        continue
+                    if limit_to == 'finished' and anime['attributes']['endDate'] is None:
+                        continue
+
+                entry = Entry()
+                entry['title'] = anime['attributes']['canonicalTitle']
+                if anime['attributes']['titles']['en']:
+                    entry['kitsu_title_en'] = anime['attributes']['titles']['en']
+                if anime['attributes']['titles']['en_jp']:
+                    entry['kitsu_title_en_jp'] = anime['attributes']['titles']['en_jp']
+                if anime['attributes']['titles']['ja_jp']:
```

Comment:
if those will not be used and are uncertain then i can also just remove thise.. the canonicalName is always going to be set as far as i can tell